### ISSUE TITLE
Build (example) error on Ubuntu 16.04.3 amd64

### ISSUE THREAD

**Issue by chetmurthy** [facebook/fbthrift#254]
I've built fbthrift's prereqs, and then fbthrift, but

(a) the default setup neither allows nor builds the example

(b) when I add what I think is the requisite line to thrift/CMakeLists.txt to build example, I get an error.
```
modified   thrift/CMakeLists.txt
@@ -9,3 +9,5 @@ if(lib_only OR build_all)
     add_subdirectory(lib)
   endif(UNIX)
 endif(lib_only OR build_all)
+
+add_subdirectory(example)
```

and the error:

```
[ 97%] Building CXX object thrift/example/cpp2/CMakeFiles/example_server.dir/server/ExampleServer.cpp.o
/home/chet/Hack/fbthrift/src/fbthrift/thrift/example/cpp2/server/ExampleServer.cpp:20:51: fatal error: proxygen/httpserver/HTTPServerOptions.h: No such file or directory
compilation terminated.
thrift/example/cpp2/CMakeFiles/example_server.dir/build.make:110: recipe for target 'thrift/example/cpp2/CMakeFiles/example_server.dir/server/ExampleServer.cpp.o' failed
make[2]: *** [thrift/example/cpp2/CMakeFiles/example_server.dir/server/ExampleServer.cpp.o] Error 1
CMakeFiles/Makefile2:1181: recipe for target 'thrift/example/cpp2/CMakeFiles/example_server.dir/all' failed
make[1]: *** [thrift/example/cpp2/CMakeFiles/example_server.dir/all] Error 2
Makefile:127: recipe for target 'all' failed
make: *** [all] Error 2
```

I'm about to go install proxygen, and see if I can't rectify this, but .... well, figured I should report it.

**Comment by chetmurthy** [comment]
I tried to continue.  Built proxygen, then found I needed to build rsocket-cpp.  So far, so good.

Now, getting a link-error:
```
ExampleServer.cpp:(.text._ZN6apache6thrift19HTTP2RoutingHandlerC2ESt10unique_ptrIN8proxygen17HTTPServerOptionsESt14default_deleteIS4_EEPNS0_15ThriftProcessorE[_ZN6apache6thrift19HTTP2RoutingHandlerC5ESt10unique_ptrIN8proxygen17HTTPServerOptionsESt14default_deleteIS4_EEPNS0_15ThriftProcessorE]+0x21): undefined reference to `vtable for apache::thrift::HTTP2RoutingHandler'
```

I'm sure I can work thru this also, but at this point, I guess I should stop and wait for some feedback.  Maybe I'm doing something completely wrong?  Maybe I should back up to some release or tag?

If it matters:

chet@twitter:~/Hack/fbthrift/src/fbthrift/build$ git log | head -4
commit 9f9aba1db1de4957caebab0360dc8d24089afc2b
Author: svcscm <svcscm@fb.com>
Date:   Sat Dec 2 18:25:54 2017 -0800

**Comment by sivabharathula** [comment]
Tried to build https://github.com/facebook/fbthrift/tree/master/thrift/lib/cpp2/transport/http2 and 
https://github.com/facebook/fbthrift/tree/master/thrift/lib/cpp2/transport/rsocket

Still not able to receive the call on the server side.

The example uses RSRoutingHandler and HTTP2RoutingHandler. Are these both required ?

**Comment by osharav** [comment]
@sivabharathula How did you manage to get the example to compile?
I also tried build and install proxygen and rsocket-cpp but the example did not compile due to the errors which @chetmurthy saw as well.

Thanks,
Omer.

**Comment by chetmurthy** [comment]
Here's what I built in order to get fbthrift working.  All git repos except zstd were on "master"; zstd was on "dev".  Also, a number of packages were downloaded as tarballs, and I don't recall if that was b/c the newer versions were incompatible with fbthrift, but that's possible.  Sorry, it was a while ago I did all this.  But I did get my own program to build and run (in addition to FBThrift examples).

git@github.com:facebook/fbthrift.git
commit 9f9aba1db1de4957caebab0360dc8d24089afc2b

folly
Fetch URL: git@github.com:facebook/folly.git
commit 7acf192db1befc5a54c8a77e2450176fe587fa1f

gflags-2.2.1
glog-0.3.5
googletest-release-1.8.0
mstch-1.0.2

proxygen
Fetch URL: git@github.com:facebook/proxygen.git
 commit cc0086e9bdb012f07b39b53c0cecc645db3b2abd

rsocket-cpp
git@github.com:rsocket/rsocket-cpp.git
commit 52965db964160d5e7d9c87f2847be560e9f4d2ac

wangle
git@github.com:facebook/wangle.git
commit bf72fc283f563f01f14ee0c232b54474a1531db2


zstd
 git@github.com:facebook/zstd.git
commit bf72fc283f563f01f14ee0c232b54474a1531db2
